java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java

```diff
@@ -272,33 +241,31 @@ private PreparedStatement generatePreparedStatement(
                     break;
             }
         }
-        if (updateDeleteValueBuffer != null) {
-            for (Object value : updateDeleteValueBuffer) {
-                stmt.setObject(placeholderIdx++, value);
-            }
-        }
-        return stmt;
     }
 
     @Override
     public void write(Iterator<SinkRow> rows) {
         while (rows.hasNext()) {
             try (SinkRow row = rows.next()) {
-                PreparedStatement stmt = prepareStatement(row);
+                Optional<PreparedStatement> stmt;
```

Comment:
We shall move the `stmt = preparedXxxStatement` to after checking `row.getOp() == Data.Op.UPDATE_DELETE`, so that the return type of preparedXxxStatement can avoid using `Optional`.